doc/conf.py

```diff
@@ -83,8 +83,17 @@
 html_theme_options = {
     "github_url": "https://github.com/numpy/numpydoc",
     "show_prev_next": False,
-    "navbar_end": ["theme-switcher", "search-field.html", "navbar-icon-links.html"],
+    "navbar_end": ["search-field.html", "navbar-icon-links.html"],
 }
+# Add version switcher for versions of pydata_sphinx_theme that support it
+import packaging
+import pydata_sphinx_theme
+
+if packaging.version.parse(pydata_sphinx_theme.__version__) >= packaging.version.parse(
+    "0.9.0"
+):
```

Comment:
This is the `conf.py` for our own docs (i.e., won't affect end users, just our own internal doc builds + deployment), so I'd rather just require 0.9.0+ from now on. WDYT?

Reply:
The problem is (I think) we use the doc build in CI as an extra set of testing, which is why I did the pinning (annoyingly) in the doc conf.

I'll try just pinning in the doc requirements - if that doesn't require a bunch of CI overhaul then I'll stick with that!

Reply:
Okay, the failing CI confirms my suspicions. The potential solutions that come to mind are:
 1. Handle the pinning/version checking dynamically in the `conf.py` as 26caf95 does
 2. Remove the jobs that build the documentation as part of the tests in CI

Other ideas? Of those two, I prefer the first as doc builds in CI can in principle help catch issues that are not explicitly covered by unit tests.

Reply:
Yeah the way you had it before is fine, I forgot about needing to stay compatible with older Sphinx etc.